src/sql/Expr.cpp

```diff
@@ -5,6 +5,39 @@
 
 namespace hsql {
 
+FrameBound::FrameBound(int64_t offset, FrameBoundType type, bool unbounded)
+    : offset{offset}, type{type}, unbounded{unbounded} {}
+
+FrameDescription::FrameDescription(FrameType type, FrameBound* start, FrameBound* end)
+    : type{type}, start{start}, end{end} {}
+
+FrameDescription::~FrameDescription() {
+  delete start;
+  delete end;
+}
+
+WindowDescription::WindowDescription(std::vector<Expr*>* partitionList, std::vector<OrderDescription*>* orderList,
+                                     FrameDescription* frameDescription)
+    : partitionList{partitionList}, orderList{orderList}, frameDescription(frameDescription) {}
```

Comment:
`frameDescription{frameDescription}` for a consistent style within the constructor.
I am not sure whether to use the newer `{}` syntax or not. (up to you)
The older `()` style is still dominant in the rest of the code.